src/server/network/drain.c

```
@@ -0,0 +1,279 @@
+    int n = encode_game(tmp, payload, len, false, 0);
+    if (n <= 0) return false;
+    bc_qmsg_t *m = msgq_push(&d->unreliable, tmp, n);
+    if (!m) return false;
+    /* Pass 3 promotes an unreliable message to reliable tracking on first
+     * delivery -- matching the stock burst behaviour where a drained
+     * unreliable message graduates into the retransmit path. */
+    m->promote = true;
+    return true;
+}
+
+bool bc_drain_enqueue_ack(bc_drain_t *d, u16 seq, u8 flags)
```

Clang (14.0.6)

[warning] parameter name 'd' is too short, expected at least 3 characters

(readability-identifier-length)